src/query/storage/m3/consolidators/id_dedupe_map.go

```diff
@@ -129,10 +136,12 @@ func (m *idDedupeMap) doUpdate(
 	}
 	if existsBetter {
 		// Existing result is already better
+		iter.Close()
 		return nil
 	}
 
 	// Override
+	existing.iter.Close()
```

Comment:
could you merge this in a separate PR? Since it's not directly related to this feature? Maybe we some testing ment to catch this?

Reply:
That's a good point, I was thinking about it as well. Did not separate initially because that is forcing me to mess up some test setup (see https://github.com/m3db/m3/pull/4003/commits/3e3722b062b5229b521227ccece39776964ad970), but I think you are right, it will be a more sane approach to submit that fix as a separate PR.